notebooks/demo.clj

```diff
@@ -1,101 +1,142 @@
 (wl/eval (w/Dot [1 2 3] [4 5 6]))
-(md "or, using our fancy aliases,")
+(k/md "or, using one of our fancy aliases,")
 (wl/eval (w/<*> [1 2 3] [4 5 6]))
-(md ".")
+(k/md ", which may be more familiar to the Mathematically inclined.
 
-(md "## Strings of WL code")
+Here, the 'w' namespace is a preconfigured, but configurable, intermediary to Wolfram's built-in symbols. This allows you to manipulate Wolfram functions just like any other clojure symbol (and get reliable editor autocompletion).")
 
-(wl/eval "{1 , 2, 3} . {4, 5, 6}")
+(k/md "## Code strings
+
+The above examples are the preferred ways to interop. You can however, use Wolfram command strings directly, e.g.")
 
-(md "## Def / intern WL fns, i.e. effectively define WL fns as clojure fns:")
+(wl/eval "{1 , 2, 3} . {4, 5, 6}")
 
-(def W:Plus (parse/parse-fn 'Plus {:jlink-instance (wolframite.impl.jlink-instance/get)}))
+(k/md "## Explicit interning
 
-(W:Plus 1 2 3) ; ... and call it
+Although it shouldn't normally be necessary, we can also intern Wolfram functions more directly i.e. to effectively define Wolfram functions as clojure functions.
 
+The standard way of doing this is something like")
 (def greetings
   (wl/eval
    (w/fn [x] (w/StringJoin "Hello, " x "! This is a Mathematica function's output."))))
-
 (greetings "Stephen")
 
-(md "## Bidirectional translation
-(Somewhat experimental, especially in the wl->clj direction)")
+(k/md "But this can also be done at a lower level, e.g.")
+(def W:Plus
+  (parse/parse-fn 'Plus {:jlink-instance (wolframite.impl.jlink-instance/get)}))
+(W:Plus 1 2 3)
+
+(k/md "## Bidirectional translation (experimental)
+
+Code translation in both directions is more difficult and is still somewhat experimental (especially in the wl->clj direction), but the basics work as expected, e.g.")
 
 (wl/->clj "GridGraph[{5, 5}]")
 (wl/->wl (w/GridGraph [5 5]) {:output-fn str})
 
-(md "## Graphics")
+(k/md "## Graphics
 
-(view (w/GridGraph [5 5]))
+The above code however, within Mathematica, actually produces graphics. Does Wolframite support this?
+
+Yes!
+")
 
-(view (w/GridGraph [5 5])
-      {:folded? true})
+(view (w/GridGraph [5 5]))
 
 (view (w/ChemicalData "Ethanol" "StructureDiagram"))
 
-(md "## More Working Examples")
+(view (w/TextStructure "The cat sat on the mat."))
+
+(k/md "The above graphics were created using the")
+^:kindly/hide-code
+(k/code "view")
+(k/md "function, as required above, and assumes that graphics are to be displayed in a browser.")
 
-(wl/eval (w/GeoNearest (w/Entity "Ocean") w/Here))
+(k/md "## Computational knowledge
 
-(md "TODO: Make this work with `view` as well.")
+Wolfram is also known for its dynamic or 'computational knowledge' engine.
 
-(view (w/TextStructure "The cat sat on the mat."))
+This can be accessed by many functions directly, ")
 
-(md "## Wolfram Alpha")
+(view (w/GeoNearest (w/Entity "Ocean") w/Here))
```

Comment:
Why don't we show this code line, and others like it? Wouldn't it be better?

Reply:
I agree. This is a consequence of the clay options. It's probably better to hide hiccup in general, but for a tutorial then probably not.